packages/wallet/src/redux/indirect-funding/player-a/state.ts

```diff
@@ -49,33 +44,50 @@ export type PlayerAState =
 export function waitForApproval(params: Properties<WaitForApproval>): WaitForApproval {
   const { channelId } = params;
 
-  return { type: WAIT_FOR_APPROVAL, player: PlayerIndex.A, channelId };
+  return { type: WAIT_FOR_APPROVAL, channelId };
 }
 
 export function waitForPreFundSetup1(
   params: Properties<WaitForPreFundSetup1>,
 ): WaitForPreFundSetup1 {
   const { channelId, ledgerId } = params;
-  return { type: WAIT_FOR_PRE_FUND_SETUP_1, player: PlayerIndex.A, channelId, ledgerId };
+  return { type: WAIT_FOR_PRE_FUND_SETUP_1, channelId, ledgerId };
 }
 
 export function waitForDirectFunding(
   params: Properties<WaitForDirectFunding>,
 ): WaitForDirectFunding {
   const { channelId, ledgerId } = params;
-  return { type: WAIT_FOR_DIRECT_FUNDING, player: PlayerIndex.A, channelId, ledgerId };
+  return { type: WAIT_FOR_DIRECT_FUNDING, channelId, ledgerId };
 }
 
 export function waitForPostFundSetup1(
   params: Properties<WaitForPostFundSetup1>,
 ): WaitForPostFundSetup1 {
   const { channelId, ledgerId } = params;
-  return { type: WAIT_FOR_POST_FUND_SETUP_1, player: PlayerIndex.A, channelId, ledgerId };
+  return { type: WAIT_FOR_POST_FUND_SETUP_1, channelId, ledgerId };
 }
 
 export function waitForLedgerUpdate1(
   params: Properties<WaitForLedgerUpdate1>,
 ): WaitForLedgerUpdate1 {
   const { channelId, ledgerId } = params;
-  return { type: WAIT_FOR_LEDGER_UPDATE_1, player: PlayerIndex.A, channelId, ledgerId };
+  return { type: WAIT_FOR_LEDGER_UPDATE_1, channelId, ledgerId };
+}
+
+// -------
+// Helpers
+// -------
+
+export function isPlayerAState(state: IndirectFundingState): state is PlayerAState {
```

Comment:
One (minor) downside is that we have another type guard that needs to be updated whenever we add/remove a state.

Reply:
That's true. I guess that you could catch that by always using these guards with exhaustive checks:
```ts
// instead of this:
if (isPlayerAState(state)) {
    // do player A stuff
} else {
    // do player B stuff
}

// do this:
if (isPlayerAState(state)) {
    // do player A stuff
} else if (isPlayerBState(state)) {
    // do player B stuff
} else {
    unreachable(state)
}
```
In fact, one test like that would ensure that the tests would catch it if you ever missed a state.

Overall it seems a bit borderline as to whether this is an improvement or not. I guess it's useful to be aware of it as an alternative.